src/main/java/com/amazonaws/encryptionsdk/EncryptRequest.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except
+ * in compliance with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.amazonaws.encryptionsdk;
+
+import com.amazonaws.encryptionsdk.keyrings.Keyring;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+
+public class EncryptRequest extends AwsCryptoRequest {
+
+    private final byte[] plaintext;
+    private final Map<String, String> encryptionContext;
+
+    private EncryptRequest(Builder builder) {
+        super(builder);
+
+        requireNonNull(builder.plaintext, "plaintext is required");
+        requireNonNull(builder.encryptionContext, "encryptionContext is required");
+        this.plaintext = builder.plaintext;
+        this.encryptionContext = builder.encryptionContext;
+    }
+
+    public byte[] plaintext() {
+        return this.plaintext;
+    }
+
+    public Map<String, String> encryptionContext() {
+        return this.encryptionContext;
+    }
```

Comment:
NP: I think it's the right call not to make defensive copies in this class. I would recommend putting a comment explaining that it's intentional. If FindBugs are used (I can't recall if they are or not) you'll also need to suppress warnings.

Reply:
Actually there is a defensive copy of encryptionContext created in the Builder. Should I take that out?

Reply:
I've added a comment for plaintext and ciphertext